### PR TITLE
pageserver: local layer path followups

### PR DESCRIPTION
**Pull request by jcsp**
- Rename "filename" types which no longer map directly to a filename (LayerFileName -> LayerName)
- Add a -v1- part to local layer paths to smooth the path to future updates (we anticipate a -v2- that uses checksums later)
- Rename methods that refer to the string-ized version of a LayerName to no longer be called "filename"
- Refactor reconcile() function to use a LocalLayerFileMetadata type that includes the local path, rather than carrying local path separately in a tuple and unwrap()'ing it later.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/tenant/timeline/init.rs [hunk 1]
```diff
@@ -80,51 +102,41 @@ pub(super) enum Decision {
...
 
 /// A layer needs to be left out of the layer map.
 #[derive(Debug)]
 pub(super) enum DismissedLayer {
     /// The related layer is is in future compared to disk_consistent_lsn, it must not be loaded.
     Future {
-        /// The local metadata. `None` if the layer is only known through [`IndexPart`].
-        local: Option<LayerFileMetadata>,
+        /// The path to local layer file. `None` if the layer is only known through [`IndexPart`].
+        local: Option<LocalLayerFileMetadata>,
```
Comment (problame) [inline pageserver/src/tenant/timeline/init.rs:118]: Still mentions path. Maybe we can remove the first sentence alltogether?

File: pageserver/src/tenant/tasks.rs [hunk 2]
```diff
@@ -375,6 +375,9 @@ async fn gc_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
                     .gc_iteration(None, gc_horizon, tenant.get_pitr_interval(), &cancel, &ctx)
                     .await;
                 if let Err(e) = res {
+                    if cancel.is_cancelled() {
+                        break;
+                    }
```
Comment (problame) [inline pageserver/src/tenant/tasks.rs:380]: This seems unrelated to the PR?

File: test_runner/regress/test_pageserver_generations.py [hunk 3]
```diff
@@ -703,7 +703,6 @@ def test_multi_attach(
     workload.validate(pageservers[2].id)
 
 
-@pytest.mark.skip(reason="To be enabled after release with new local path style")
```
Comment (problame) [inline test_runner/regress/test_pageserver_generations.py:706]: Just making sure you're not slipping this through accidentally.

Comment (jcsp) [reply test_runner/regress/test_pageserver_generations.py]: oops, that should have been reinstated in the last commit.  bbd7e50e3

File: pageserver/src/tenant/tasks.rs (hunk 2 above)
Comment (jcsp) [reply pageserver/src/tenant/tasks.rs]: oops.  Harmless but unintended for this PR.  49666770f

File: pageserver/src/tenant/timeline/init.rs (hunk 1 above)
Comment (jcsp) [reply pageserver/src/tenant/timeline/init.rs]: Cleaned up in a1d43d8df
